server.py

```diff
@@ -57,12 +57,17 @@ async def create_url(request: Request):
     except ValueError:
         logging.exception(f"returning 422 due to invalid alias of \"{alias}\"")
         raise HTTPException(status_code=HttpResponse.INVALID_ARGUMENT_EXCEPTION.code)
-   
+
 @app.get("/list")
-async def get_all_urls():
+async def get_urls(search: Optional[str] = None, page: int = 0):
+    if page < 0:
+        raise HTTPException(status_code=400, detail="Invalid page number")
+    if search and not search.isalnum():
+        raise HTTPException(status_code=400, detail=f'search term "{search}" is invalid. only alphanumeric chars are allowed')
     with MetricsHandler.query_time.labels("list").time():
-      return sqlite_helpers.get_urls(DATABASE_FILE)
-
+        urls = sqlite_helpers.get_urls(DATABASE_FILE, page, search=search)
+        total_urls = sqlite_helpers.get_number_of_entries(DATABASE_FILE)
```

Comment:
i think we also need to pass the search in here, otherwise we will get an incorrect total?

Reply:
testing discord webhooks

Reply:
done in 68431a721993c2770e4508b097fd4d7127fa892e